src/core/lib/transport/metadata_batch.h

```diff
@@ -314,13 +376,15 @@ struct SimpleIntBasedMetadata : public SimpleIntBasedMetadataBase<Int> {
 struct GrpcStatusMetadata
     : public SimpleIntBasedMetadata<grpc_status_code, GRPC_STATUS_UNKNOWN> {
   static constexpr bool kRepeatable = false;
+  using CompressionTraits = SmallIntegralValuesCompressor<16>;
   static absl::string_view key() { return "grpc-status"; }
 };
 
 // grpc-previous-rpc-attempts metadata trait.
 struct GrpcPreviousRpcAttemptsMetadata
     : public SimpleIntBasedMetadata<uint32_t, 0> {
   static constexpr bool kRepeatable = false;
+  using CompressionTraits = SmallIntegralValuesCompressor<3>;
```

Comment:
Could you confirm this needs to be SmallIntegralValuesCompressor instead of NoCompression ? Maybe I missed it but previous code doesn't seem to do any compression for this. Also is "3" arbitrary here ?

Reply:
you're right, typo.